### PR TITLE
load_config_file: Note `--save` on no config

### PR DESCRIPTION
**Pull request by sils1297**
Fixes https://github.com/coala-analyzer/coala/issues/2241

### CODE REVIEW

**Comment by sils1297** [comment]
ack https://github.com/coala-analyzer/coala/pull/2246/commits/9cacce0a3aa19ac08646fae87f13a8287c55e105

File: coalib/settings/ConfigurationGathering.py [hunk 1]
```diff
@@ -56,12 +56,14 @@ def load_config_file(filename, log_printer, silent=False):
     except FileNotFoundError:
         if not silent:
             if os.path.basename(filename) == Constants.default_coafile:
-                log_printer.warn("The default coafile " +
-                                 repr(Constants.default_coafile) + " was not "
-                                 "found. Ignoring it.")
+                log_printer.warn("The default coafile {0!r} was not found. "
+                                 "You can generate a configuration from "
+                                 "command line options with the `--save` flag."
```
Comment (underyx) [inline coalib/settings/ConfigurationGathering.py]: How about
> You can generate one with your current options by adding the `--save` flag.
or
> You can generate a configuration file with your current options by adding the `--save` flag.

File: coalib/settings/ConfigurationGathering.py [hunk 2]
```diff
@@ -56,12 +56,14 @@ def load_config_file(filename, log_printer, silent=False):
...
+                                 .format(Constants.default_coafile))
             else:
-                log_printer.err("The requested coafile " + repr(filename) +
-                                " does not exist.")
+                log_printer.err("The requested coafile {0!r} does not exist"
+                                ". You can generate it with `--save {0!r}`."
```
Comment (underyx) [inline coalib/settings/ConfigurationGathering.py]: How about

>You can generate it with your current options by adding the --save flag.

File: tests/coalaJSONTest.py [hunk 3]
```diff
@@ -60,7 +60,7 @@ def test_text_logs(self):
             coala_json.main, 'coala-json', '--text-logs', '-c', 'nonex')
         self.assertRegex(
             output,
-            ".*\\[ERROR\\].*The requested coafile '.*' does not exist.\n")
+            ".*\\[ERROR\\].*The requested coafile '.*' does not exist. .+\n")
```
Comment (underyx) [inline tests/coalaJSONTest.py]: I think

>`".*\\[ERROR\\].*The requested coafile '.*' does not exist. .+\n"`

should be equal to

>`r".*\[ERROR\].*The requested coafile '.*' does not exist. .+\n"`

with the latter being more readable.

**Comment by sils1297** [comment]
reack https://github.com/coala-analyzer/coala/pull/2246/commits/09868135fbb5c1fac38ef35526c47a446d051720

File: coalib/settings/ConfigurationGathering.py [hunk 4]
```diff
@@ -56,12 +56,15 @@ def load_config_file(filename, log_printer, silent=False):
...
+                log_printer.warn("The default coafile {0!r} was not found. You"
+                                 " can generate a configuration file with your"
+                                 " current options by adding the --save flag."
+                                 .format(Constants.default_coafile))
             else:
-                log_printer.err("The requested coafile " + repr(filename) +
-                                " does not exist.")
+                log_printer.err("The requested coafile {0!r} does not exist. "
+                                "You can generate it with your current "
+                                "options by adding the `--save` flag."
```
Comment (underyx) [inline coalib/settings/ConfigurationGathering.py]: Now this one says
```
`--save`
```

and the other says just

```
--save
```

I don't have an opinion on which one to use, but they should definitely be consistent.

**Comment by underyx** [comment]
unack https://github.com/coala-analyzer/coala/pull/2246/commits/7ba3d44ee4de3ea08763cb1a653675f8c5db27ac

**Comment by underyx** [comment]
ack https://github.com/coala-analyzer/coala/pull/2246/commits/0bc8e50269fd4803976fd8a8bb2eedec58f2c10d

**Comment by sils1297** [comment]
reack a6d45bf

**Comment by sils1297** [comment]
@rultor merge

**Pull request merged**